File: CodeChef/Tom_and_Jerry_1.py

t = int(input())

for _ in range(t):
    a,b,c,d,k = map(int,input().split())
    p1_parity = sum((a,b))
    p2_parity = sum((c,d))

    dist = abs(c-a)+abs(d-b)

    if(k<dist):
        print("NO")
    else:
        if((p1_parity%2 ==0 and p2_parity%2==0) or (p1_parity%2 !=0 and p2_parity%2 !=0)):
            if(k%2==0):
                print("YES")
            else:
                print("NO")
        else:
            if(k%2!=0):
                print("YES")
            else:
                print("NO")

File: CodeChef/Tom_and_Jerry_1.cpp
#include<std/c++11.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int a, b, c, d, k;
        cin >> a >> b >> c >> d >> k;

        int dist = abs(c-a) + abs(d-b);
        if (k >= dist && k %2 == dist % 2)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}